src/nvim/ex_cmds.c

```diff
@@ -5836,3 +5873,166 @@ void set_context_in_sign_cmd(expand_T *xp, char_u *arg)
+  if (cmdwin_type != 0
+      || cmdline_star > 0
+      ) {
+    beep_flush();
+    return K_IGNORE;
+  }
+
+  // Save current window sizes. 
+  win_size_save(&winsizes);
+
+  // Save the current window to restore it later 
+  win_T* oldwin = curwin;
+
+  // Don't execute autocommands while creating the window. 
+  block_autocmds();
+  // don't use a new tab page 
+  cmdmod.tab = 0;
+
+  // close last buffer used for ex_window_live_sub() 
+  buf_T* oldbuf;
+  if((oldbuf = buflist_findname_exp((char_u *)"[live_sub]"))!=NULL) {
+    close_windows (oldbuf, FALSE);
+    close_buffer (NULL, oldbuf, DOBUF_WIPE, FALSE);
+  }
+
+  // Create a window for the command-line buffer. 
+  if (win_split((int)p_cwh, WSP_BOT) == FAIL) {
+    beep_flush();
+    unblock_autocmds();
+    return K_IGNORE;
+  }
+  cmdwin_type = get_cmdline_type();
+
+  // Create the command-line buffer empty. 
+  (void)do_ecmd(0, NULL, NULL, NULL, ECMD_ONE, ECMD_HIDE, NULL);
+  (void)setfname(curbuf, (char_u *)"[live_sub]", NULL, TRUE);
+  set_option_value((char_u *)"bt", 0L, (char_u *)"nofile", OPT_LOCAL);
+  set_option_value((char_u *)"swf", 0L, NULL, OPT_LOCAL);
+  curbuf->b_p_ma = FALSE; // Not Modifiable
+  curwin->w_p_fen = FALSE;
+  curwin->w_p_rl = cmdmsg_rl;
+  cmdmsg_rl = FALSE;
+  RESET_BINDING(curwin);
+
+  // Do execute autocommands for setting the filetype (load syntax). 
+  unblock_autocmds();
+
+  // Showing the prompt may have set need_wait_return, reset it. 
+  need_wait_return = FALSE;
+
+  // Reset 'textwidth' after setting 'filetype' (the Vim filetype plugin sets 'textwidth' to 78). 
+  curbuf->b_p_tw = 0;
+
+  // Initialize line and highliht variables 
+  int line = 0;
+  int src_id_highlight = 0;
+  long match_size = strlen((char*)sub);
+  
+  // allocate a line sized for the window
+  char *str = xmalloc((size_t )curwin->w_frame->fr_width);
+  
+  // Append the lines to our buffer
+  kl_iter(matchedline_T, lmatch, current) {
+    matchedline_T mat = (*current)->data;
+    size_t line_size = sizeof(mat.line) + sizeof(long) + 5;
+
+    // Reallocation if str not long enough
+    if (line_size > curwin->w_frame->fr_width*sizeof(char))
+      assert(xrealloc(str, line_size) != NULL);
+
+    // Add the line number to the string
+    // TODO : enhance the display of the line number 
+    sprintf(str, "l.%ld > %s", mat.lnum, (char*)mat.line);
+    ml_append(line++, (char_u *)str, (colnr_T)0, false);
+    
+    int prefix_size = strlen(str) - strlen((char*)mat.line);
+    
+    kl_iter(colnr_T, mat.start_col, col) {
+      src_id_highlight = bufhl_add_hl(curbuf, 
+                                      src_id_highlight,
+                                      curbuf->handle,
+                                      line,                                     // line in curbuf
+                                      (*col)->data + prefix_size + 1,           // beginning of word
+                                      (*col)->data + prefix_size + match_size); // end of word
```

Comment:
What if there is more than one match per line and the sub has a different length to what is replaced (`:%s/e/XXXX/g`)?